src/proxy.ts

```diff
@@ -22,8 +22,18 @@ function getDeploymentAddresses(chainId: ethers.BigNumberish) {
   return { api3ServerV1Address, proxyFactoryAddress };
 }
 
-function computeDapiProxyAddress(chainId: ethers.BigNumberish, dapiName: string, metadata: ethers.BytesLike) {
-  const { api3ServerV1Address, proxyFactoryAddress } = getDeploymentAddresses(chainId);
+interface ContractAddresses {
+  api3ServerV1Address: string;
+  proxyFactoryAddress: string;
+}
+
+function computeDapiProxyAddress(
+  chainId: ethers.BigNumberish,
+  dapiName: string,
+  metadata: ethers.BytesLike,
+  contracts: ContractAddresses = getDeploymentAddresses(chainId)
```

Comment:
If one specifies `contracts`, `chainId` becomes redundant, so I'm not sure that I like this (or I misunderstood the syntax)

Reply:
I think one 
```
function computeDapiProxyAddress(
  chainId: ethers.BigNumberish,
  dapiName: string,
  metadata: ethers.BytesLike
  )
```
and one
```
function computeDapiProxyAddress(
  contracts: ContractAddresses,
  dapiName: string,
  metadata: ethers.BytesLike
  )
```
makes more sense but the second one is more error-prone and I don't see who would need it

Reply:
> If one specifies contracts, chainId becomes redundant, so I'm not sure that I like this (or I misunderstood the syntax)

You've understood it well. And, yeah the `chainId` is redundant in that case. I wanted my change to be backwards compatible with existing usage so it felt natural to me to make the "override" as a second argument.

> I think one ... and one ...

The problem with that is, you can't overload functions in JS, so this would end up as
```
function computeDapiProxyAddress(
  chainIdOrContracts: ethers.BigNumberish | ContractAddresses,
  dapiName: string,
  metadata: ethers.BytesLike
  )
```

which I don't like.

> I don't see who would need it

For example, we have e2e tests for auctioneer and use custom chains, for which there are no deployments in @api3/contracts. I'd be OK if we only had the second variant and called `getDeploymentAddresses(chainId)` in the auctioneer.